arm-servicefabric/2016-09-01/swagger/servicefabric.json

```diff
@@ -0,0 +1,757 @@
+        "clientCertificateThumbprints": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ClientCertificateThumbprint"
+          }
+        },
+        "clientCertificateCommonNames": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ClientCertificateCommonName"
+          }
+        },
+        "fabricSettings": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SettingsSectionDescription"
+          }
+        },
+        "reverseProxyCertificate": {
+          "$ref": "#/definitions/CertificateDescription"
+        },
+        "managementEndpoint": {
+          "type": "string",
+          "readOnly": true
+        },
+        "nodeTypes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NodeTypeDescription"
+          }
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Default",
+            "WaitingForNodes",
+            "Deploying",
+            "BaselineUpgrade",
+            "UpdatingUserConfiguration",
+            "UpdatingUserCertificate",
+            "UpdatingInfrastructure",
+            "EnforcingClusterVersion",
+            "UpgradeServiceUnreachable",
+            "Deleting",
+            "ScaleUp",
+            "ScaleDown",
+            "AutoScale",
+            "Ready",
+            "Failed"
+          ]
+        },
+        "vmImage": {
+          "type": "string",
+          "readOnly": true
+        },
+        "diagnosticsStorageAccountConfig": {
+          "$ref": "#/definitions/diagnosticsStorageAccountConfig"
+        },
+        "upgradeDescription": {
+          "$ref": "#/definitions/paasClusterUpgradePolicy"
+        }
+      }
+    },
+    "clusterPatchParameters": {
+      "description": "Cluster patch parameters",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/clusterResourceProperties"
+        },
+        "propertyNames": {
+          "description": "Allowed properties",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "enum": [
+              "ReliabilityLevel",
+              "ClusterUpgradeMode",
+              "ClientCertificateThumbprints",
+              "FabricSettings",
+              "ReverseProxyCertificate",
+              "NodeTypes"
+            ]
+          }
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "x-ms-azure-resource": false
+    },
+    "ClusterPatchRequest": {
+      "description": "Cluster patch request",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/clusterPatchParameters"
+        }
+      }
+    },
+    "ClusterResource": {
+      "description": "The cluster resource",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/clusterResourceProperties"
+        }
+      },
```

Comment:
By default, we flatten 'properties' by one level. Here in this case 

```json
ClusterResource: {
  "properties": {
    "firstProperty": {
       "type": "string"
    },
    "properties": {
     "$ref": "#/definitions/clusterResourceProperties"
    }
  }
}
```

The generated class will be used as:

```java
ClusterResource rs;
rs.myProperty = "";
rs.properties.clusterId = "";
```

But if you apply `x-ms-client-flatten` extension https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-extensions.md#x-ms-client-flatten as below:

```json
ClusterResource: {
  "properties": {
    "firstProperty": {
       "type": "string"
    },
    "properties": {
     "x-ms-client-flatten": true,
     "$ref": "#/definitions/clusterResourceProperties"
    }
  }
}
```

then generated type will be used as:

```java
ClusterResource rs;
rs.myProperty = "";
rs.clusterId = "";
```

We recommend using this extension so that end users experience will be better. I guess this is one of the warning you are getting from validator.